### PR TITLE
Feature(reviews): Add hasSpoiler Checkbox

### PR DESCRIPTION
**Pull request by Vicentesan**
# This PR introduces a new feature that allows users to mark a review as containing spoilers. The following changes were made:

## "Contains Spoiler" Checkbox:

- Added a checkbox to the review form that users can select if their review contains spoilers.

## Database Schema Update:

- New column contains_spoiler added to the reviews table, as a boolean field (default: false). - Still need to create the migration file

## Updated Create Review Function:

- Modified the function responsible for creating a review to include the contains_spoiler flag.
Now, when a review is created, the contains_spoiler value from the form is properly saved in the database.

## Changes:

- Frontend:
  - Added the "Contains Spoiler" checkbox to the review submission form.
  - Adjusted the review display logic to handle spoiler reviews accordingly.
  
- Backend/Functions:
  - Modified create review function to accept and process the contains_spoiler flag.

- Database:
  - Added the "contain_spoilers" field in the schema
  
## Screenshots:
Before:
image

After:
image

# Testing:
 - [ ❌] Tested the review submission process with and without the spoiler checkbox checked.
 - [ ❌] Verified the database correctly stores and retrieves the contains_spoiler field.
 - [ ❌] Checked the frontend displays reviews with spoilers appropriately.
 - [ ❌] Tested the review creation function to ensure it properly saves the contains_spoiler flag.
 
 I'll do all these tests when I properly migrate the dev database
 
 ## Todo:
- Add modal warning when the user attempts to publish the review without checking the checkbox to make sure he didn't forgot about it
 
Related Issues:
https://github.com/plotwist-app/plotwist/issues/234

### CODE REVIEW

File: apps/web/src/services/api/reviews/create-review.ts [hunk 1]
```diff
@@ -6,13 +7,15 @@ export const createReview = async ({
   userId,
   mediaType,
   rating,
+  contain_spoilers,
```
Comment (eulixir) [inline apps/web/src/services/api/reviews/create-review.ts:10]: Eu trocaria para `has_spoilers`, o que acha?. E pode deixar o campo camelCase, o prisma lida com essa conversão.

Comment (Vicentesan) [reply apps/web/src/services/api/reviews/create-review.ts]: alterado no commit a seguir

File: apps/web/src/services/api/reviews/create-review.ts [hunk 2]
```diff
@@ -6,13 +7,15 @@ export const createReview = async ({
...
   rating,
+  hasSpoilers,
   review,
   tmdbItem,
   language,
 }: CreateReviewValues) => {
   const tmdbItemValues = sanitizeTmdbItem(tmdbItem)
 
   const { error, data } = await supabase.from('reviews').insert({
+    hasSpoilers,
```
Comment (eulixir) [inline apps/web/src/services/api/reviews/create-review.ts:18]:
```suggestion
    has_spoiler: hasSpoilers,
```

File: apps/web/src/services/api/reviews/create-review.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable camelcase */
```
Comment (eulixir) [inline apps/web/src/services/api/reviews/create-review.ts:1]:
```suggestion
```

File: debug.log [hunk 4]
```diff
@@ -0,0 +1 @@
+[1023/222729.906:ERROR:crashpad_client_win.cc(810)] not connected
```
Comment (eulixir) [inline debug.log:1]:
```suggestion
```

avoid commit the debug log to github

**Review by eulixir — CHANGES_REQUESTED** [review]
Do you think it's a good idea to put a blur when the review is marked as hasSpoiler == true?

File: apps/web/src/services/api/reviews/create-review.ts (hunk 2 above)
Comment (Vicentesan) [reply apps/web/src/services/api/reviews/create-review.ts]: changed in the next commit

File: apps/web/src/services/api/reviews/create-review.ts (hunk 3 above)
Comment (Vicentesan) [reply apps/web/src/services/api/reviews/create-review.ts]: removed in the next commit

File: debug.log (hunk 4 above)
Comment (Vicentesan) [reply debug.log]: removed in the next commit

**Comment by Vicentesan** [comment]
> Do you think it's a good idea to put a blur when the review is marked as hasSpoiler == true?

Yes, I do think its a good Idea. I will do it now!

**Comment by Vicentesan** [comment]
@eulixir Ready for merge. Last review please?

**Review by eulixir — APPROVED** [review]
LGTM, waiting for @lui7henrique review to merge

**Pull request merged**